itou/job_applications/migrations/0010_set_jobapplication_archived_at.py

```diff
@@ -0,0 +1,60 @@
+import datetime
+import time
+
+from django.db import migrations
+from django.utils import timezone
+
+
+def forward(apps, editor):
+    JobApplication = apps.get_model("job_applications", "JobApplication")
+    BATCH_SIZE = 10_000
+    now = timezone.now()
+    six_months_ago = now - datetime.timedelta(days=180)
+    total = 0
+    start = time.perf_counter()
+
+    archivable_states = ["new", "processing", "postponed", "refused", "cancelled", "obsolete"]
+    print()
+    while chunk := JobApplication.objects.filter(
+        state__in=archivable_states,
+        updated_at__lte=six_months_ago,
+        archived_at=None,
```

Comment:
Attention aux filtres sur des champs sans index : il vaut mieux extraire la liste des ids des candidatures à traiter, puis itérer dessus.
Voir https://itou-inclusion.slack.com/archives/C0412CTV63D/p1725541863233409?thread_ts=1725526366.771849&cid=C0412CTV63D
et https://github.com/gip-inclusion/les-emplois/pull/4662/files#diff-52d271ec0e4d9971fc1a02420ea32c7e1711afe3512aa8a92d7745526bcef79f

Reply:
Effectivement, il reste bien plus efficace d’identifier d’abord. La version qui identifie à chaque itération passait en 1200s sur ma machine, la version de tout identifier d’un coup et faire des batchs prend 300s. La requête pour tout identifier est un peu lourde (10s), mais je ne pense pas utile de la découper en buckets de dates.